client/lib/query-manager/theme/index.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * External dependencies
+ */
+import { every, some, includes, startsWith } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import PaginatedQueryManager from '../paginated';
+import ThemeQueryKey from './key';
+import { DEFAULT_THEME_QUERY } from './constants';
+
+/**
+ * ThemeQueryManager manages themes which can be queried
+ */
+export default class ThemeQueryManager extends PaginatedQueryManager {
+	/**
+	 * Returns true if the theme matches the given query, or false otherwise.
+	 *
+	 * @param  {Object}  query Query object
+	 * @param  {Object}  theme Item to consider
+	 * @return {Boolean}       Whether theme matches query
+	 */
+
+	matches( query, theme ) {
+		const queryWithDefaults = Object.assign( {}, DEFAULT_THEME_QUERY, query );
+		return every( queryWithDefaults, ( value, key ) => {
+			switch ( key ) {
+				case 'search':
+					if ( ! value ) {
+						return true;
+					}
+
+					const search = value.toLowerCase();
+
+					const taxonomiesToSearch = [ 'subject', 'feature', 'color', 'style', 'column', 'layout' ];
```

Comment:
Maybe bump to top-level constant `SEARCH_TAXONOMIES` to avoid redeclaring each time?